sound/soc/codecs/rt5682.c

```diff
@@ -995,6 +995,14 @@ static int rt5682_set_jack_detect(struct snd_soc_component *component,
 {
 	struct rt5682_priv *rt5682 = snd_soc_component_get_drvdata(component);
 
```

Comment:
We need to set rt5682->hs_jack = hs_jack; even hs_jack = NULL to prevent rt5682->hs_jack from being garbage after codec driver is removed. So I suggest to move ``rt5682->hs_jack = hs_jack;`` from line 1043 to here.

Reply:
ok